infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/CorfuLogReplicationRuntime.java

```diff
@@ -154,19 +160,26 @@ public class CorfuLogReplicationRuntime {
     @Getter
     public final String remoteClusterId;
 
+    @Getter
+    public final LogReplicationSession session;
+
+    public final LogReplicationContext context;
+
     /**
      * Default Constructor
      */
     public CorfuLogReplicationRuntime(LogReplicationRuntimeParameters parameters,
-                                      LogReplicationMetadataManager metadataManager,
-                                      LogReplicationUpgradeManager upgradeManager,
-                                      LogReplicationContext replicationContext,
-                                      ReplicationSession replicationSession) {
-        this.remoteClusterId = replicationSession.getRemoteClusterId();
+                                      LogReplicationMetadataManager metadataManager, LogReplicationUpgradeManager upgradeManager,
+                                      LogReplicationSession session, LogReplicationContext context) {
+        this.remoteClusterId = session.getSinkClusterId();
+        this.session = session;
+        this.context = context;
```

Comment:
can we rename this to replicationContext? 
becomes clean to read.

Reply:
renamed and also removed it as an instance variable as it is only used for passing down to SourceManager.